langchain/src/agents/tools/base.ts

```diff
@@ -17,23 +17,32 @@ export abstract class Tool {
     this.callbackManager = callbackManager ?? getCallbackManager();
   }
 
-  protected abstract _call(arg: string): Promise<string>;
-
-  async call(arg: string, verbose?: boolean): Promise<string> {
-    const _verbose = verbose ?? this.verbose;
-    await this.callbackManager.handleToolStart(
+  protected abstract _call(
+    arg: string,
+    callbackManager?: CallbackManager
+  ): Promise<string>;
+
+  async call(
+    arg: string,
+    verbose?: boolean,
+    callbackManager?: CallbackManager
+  ): Promise<string> {
+    const verbose_ = verbose ?? this.verbose;
+    const callbackManager_ = callbackManager ?? this.callbackManager;
```

Comment:
To ensure this is all working correctly I'd actually completely remove this.callbackManager. Thoughts?

Reply:
yeah probably a good idea